### ISSUE TITLE
Unable to install Tensorflow - Windows 10

### ISSUE THREAD

**Issue by LuisTang** [rstudio/tensorflow#304]
Hi everyone,

I'm new to RStudio on Windows 10 and wanted to install Tensorflow in it to make use of my GPU. However, after running the command:

`devtools::install_github("rstudio/tensorflow")`

the console throws the following message:

```
Downloading GitHub repo rstudio/tensorflow@master
√  checking for file 'D:\Users\Luis Tang\AppData\Local\Temp\RtmpYBE2wk\remotes3b6454fe1812\rstudio-tensorflow-7b0babd/DESCRIPTION' ... 
-  preparing 'tensorflow':
√  checking DESCRIPTION meta-information ... 
-  checking for LF line-endings in source and make files and shell scripts
-  checking for empty or unneeded directories
-  building 'tensorflow_1.10.0.9000.tar.gz'
   
* installing *source* package 'tensorflow' ...
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
  converting help for package 'tensorflow'
    finding HTML links ... done
    all_dims                                html  
    evaluate                                html  
    export_savedmodel                       html  
    install_tensorflow                      html  
    install_tensorflow_extras               html  
    parse_arguments                         html  
    finding level-2 HTML links ... done

    parse_flags                             html  
    reexports                               html  
Rd warning: D:/Users/Luis Tang/AppData/Local/Temp/RtmpaaGgDI/R.INSTALL3760267c79/tensorflow/man/reexports.Rd:29: file link '%as%' in package 'reticulate' does not exist and so has been treated as a topic
Rd warning: D:/Users/Luis Tang/AppData/Local/Temp/RtmpaaGgDI/R.INSTALL3760267c79/tensorflow/man/reexports.Rd:29: file link 'use_virtualenv' in package 'reticulate' does not exist and so has been treated as a topic
Rd warning: D:/Users/Luis Tang/AppData/Local/Temp/RtmpaaGgDI/R.INSTALL3760267c79/tensorflow/man/reexports.Rd:29: file link 'use_condaenv' in package 'reticulate' does not exist and so has been treated as a topic
Rd warning: D:/Users/Luis Tang/AppData/Local/Temp/RtmpaaGgDI/R.INSTALL3760267c79/tensorflow/man/reexports.Rd:31: file link 'flag_numeric' in package 'tfruns' does not exist and so has been treated as a topic
Rd warning: D:/Users/Luis Tang/AppData/Local/Temp/RtmpaaGgDI/R.INSTALL3760267c79/tensorflow/man/reexports.Rd:31: file link 'flag_integer' in package 'tfruns' does not exist and so has been treated as a topic
Rd warning: D:/Users/Luis Tang/AppData/Local/Temp/RtmpaaGgDI/R.INSTALL3760267c79/tensorflow/man/reexports.Rd:31: file link 'flag_string' in package 'tfruns' does not exist and so has been treated as a topic
Rd warning: D:/Users/Luis Tang/AppData/Local/Temp/RtmpaaGgDI/R.INSTALL3760267c79/tensorflow/man/reexports.Rd:31: file link 'flag_boolean' in package 'tfruns' does not exist and so has been treated as a topic
    shape                                   html  
    sub-.tensorflow.tensor                  html  
    tensorboard                             html  
    tensorflow                              html  
    tf                                      html  
    tf_config                               html  
    tf_extract_opts                         html  
    tf_probability                          html  
    tfe_enable_eager_execution              html  
    train                                   html  
    train_and_evaluate                      html  
    use_compat                              html  
    use_session_with_seed                   html  
    view_savedmodel                         html  
** building package indices
** testing if installed package can be loaded
*** arch - i386
Error fatal: no fue posible abrir el archivo 'D:\Users\Luis': No such file or directory

*** arch - x64
Error fatal: no fue posible abrir el archivo 'D:\Users\Luis': No such file or directory

ERROR: loading failed for 'i386', 'x64'
* removing 'D:/Program Files/R/R-3.5.2/library/tensorflow'
In R CMD INSTALL
Error in i.p(...) : 
  (converted from warning) installation of package ‘D:/Users/Luis Tang/AppData/Local/Temp/RtmpYBE2wk/file3b641ab0855/tensorflow_1.10.0.9000.tar.gz’ had non-zero exit status
In addition: Warning messages:
1: In untar2(tarfile, files, list, exdir) :
  skipping pax global extended headers
2: In untar2(tarfile, files, list, exdir) :
  skipping pax global extended `headers`
```
I have already installed the following packages:

- Rcpp
- Rtools 35
- Devtools
- Tensorflow for Python

Am I missing some other software or package before I can install Tensorflow?

**Comment by jcrodriguez1989** [comment]
Hi Luis,
First, I could not reproduce your issue because I don't have Windows installed.
Could you try installing tensorflow through CRAN?
`install.packages("tensorflow")` instead of using `install_github("rstudio/tensorflow")`.

I am pretty sure this is a devtools issue. Have you tried installing any other package with `install_github()`?
I think it's a devtools issue because you are having this error:

> Error fatal: no fue posible abrir el archivo **'D:\Users\Luis'**: No such file or directory

While your user is "Luis Tang", and thus, your home dir 'D:\Users\Luis Tang'.
Another workaround would be to create a new user in Windows like "LuisTang" (no space), and try the `install_github()` with this user.

**Comment by LuisTang** [comment]
Thanks for your reply! By following your methodology I've discovered that apparently my R version is "too new" for Tensorflow:

`> install.packages("tensorflow")
Installing package into ‘D:/Users/Luis Tang/Documents/R/win-library/3.3’
(as ‘lib’ is unspecified)
Warning in install.packages :
  package ‘tensorflow’ is not available (for R version 3.3.2)`

Do you happen to know what versions of R Tensorflow is actually available for?

Thanks in advance!

**Comment by skeydan** [comment]
> apparently my R version is "too new" 

On the contrary, `tensorflow` works fine with the newest R `3.5.2`. And please at the moment do not install from CRAN, but the development version from github, like so:

```
devtools::install_github("rstudio/tensorflow") 
devtools::install_github("rstudio/keras") 
```

Thanks!
